hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/ErrorTranslation.java

```diff
@@ -106,6 +110,24 @@ public static IOException maybeExtractIOException(String path, Throwable thrown)
 
   }
 
+  /**
+   * Extracts the underlying exception from an S3EncryptionClientException.
+   * @param exception amazon exception raised
+   * @return extractedException
+   */
+  public static SdkException maybeExtractSdkException(SdkException exception) {
+    SdkException extractedException = exception;
+    if (exception.toString().contains(ENCRYPTION_CLIENT_EXCEPTION)) {
```

Comment:
1. should this be in the classname, or is it one of those things which gets passed down as strings?
2. also include check for cause being instanceof SdkException so class cast problems don't lose stack trace of any other problem